src/coreclr/vm/clsload.cpp

```diff
@@ -1238,18 +1243,18 @@ BOOL ClassLoader::FindClassModuleThrowing(
 
     if (pBucket == NULL)
     {
+        // Take the lock. To make sure the table is not being built by another thread.
         AvailableClasses_LockHolder lh(this);
 
-        // Try again with the lock.  This will protect against another thread reallocating
-        // the hash table underneath us
-        GetClassValue(nhTable, pName, &Data, &pTable, pLookInThisModuleOnly, &foundEntry, loadFlag, needsToBuildHashtable);
-        pBucket = foundEntry.GetClassHashBasedEntryValue();
-
+        if (!needsToBuildHashtable || (m_cUnhashedModules == 0))
```

Comment:
We do not need to do lookup 3 times if tables were populated by another thread when we did a lookup. 
Just one retry is sufficient in all cases.